### ISSUE TITLE
wait_for_mysql fails due to error

### ISSUE THREAD

**Issue by m4tze** [maxking/docker-mailman#358]
There are two problems with the current docker image and the wait_for_mysql method in docker-entrypoint.sh (also applies to docker-web image and maybe others).

First one is that the mysql installation is missing some library.

```
ERROR 2059 (HY000): Plugin caching_sha2_password could not be loaded: Error loading shared library /usr/lib/mariadb/plugin/caching_sha2_password.so: No such file or directory
```

I could solve the problem with ```apk update && apk add mysql``` in that container manually.

Unfortunately, the check still fails because the hostname determination is wrong (error is caused by the quoting of $DATABASE_URL). If you omit the single quotes or use them twice, the check works. This is caused by the reason that bash does not expand variables in single quotes.

Failed:
```
bash-4.4# HOSTNAME=$(python3 -c "from urllib.parse import urlparse; o = urlparse('$DATABASE_URL'); print(o.hostname);")
bash-4.4# echo $HOSTNAME

```

Working:
```
bash-4.4# HOSTNAME=$(python3 -c "from urllib.parse import urlparse; o = urlparse($DATABASE_URL); print(o.hostname);")
bash-4.4# echo $HOSTNAME
database
```

also working:
```
bash-4.4# HOSTNAME=$(python3 -c "from urllib.parse import urlparse; o = urlparse(''$DATABASE_URL''); print(o.hostname);")
bash-4.4# echo $HOSTNAME
database
```

Is it necessary to create an issue for the docker-web (and maybe other related projects)?

Kind regards
- Matze

**Comment by maxking** [comment]
Thanks for the report, 2nd one clearly is a bug. Would you like to submit a MR to fix that? :)

I am not sure which library provides the expected .so, but I presume `apk` might have some API to search through. I'll look into figuring out the missing dependency.

**Comment by m4tze** [comment]
Hi maxking,

I found out, that this bug only appears when using mysql:latest as an image for the mysql server. Using mariadb:10.3, as suggested in the docker-compose-mysql.yaml, everything seems to work. 

For the second problem: This also vanishes when using mariadb, it turned out that I quoted the DATABASE_URL string in the docker-compose.yaml to find the error with the mysql:latest image. (I haven't found you in the #mailman IRC so I started to investigate on my own ;) thanks for helping me out.)

I would have posted an MR but I think the only way to handle this is closing this issue ;)

Kind regards,
- Matze
